src/API/Google/MerchantPriceBenchmarks.php

```diff
@@ -77,8 +94,28 @@ public function get_price_insights( array $args ): array {
 			->set_client( $this->service, $this->options->get_merchant_id() )
 			->get_results();
 
-			$results = $response->getResults() ?? [];
-
+			$price_insights_data = $response->getResults() ?? [];
+
+			if ( empty( $price_insights_data ) ) {
+				return $price_insights_data;
+			}
+
+			// Map the benchmark data to a require format.
+			$results = [];
+			foreach ( $price_insights_data as $price_insights_result ) {
+				$results[] = [
+					'id'                               => $price_insights_result->getProductView()->getId(),
+					'offer_id'                         => $price_insights_result->getProductView()->getOfferId(),
+					'title'                            => $price_insights_result->getProductView()->getTitle(),
+					'price_micros'                     => $price_insights_result->getProductView()->getPriceMicros(),
+					'currency_code'                    => $price_insights_result->getProductView()->getCurrencyCode(),
+					'suggested_price_micros'           => $price_insights_result->getPriceInsights()->getSuggestedPriceMicros(),
+					'predicted_impressions_change_fraction' => $price_insights_result->getPriceInsights()->getPredictedImpressionsChangeFraction(),
+					'predicted_clicks_change_fraction' => $price_insights_result->getPriceInsights()->getPredictedClicksChangeFraction(),
+					'predicted_conversions_change_fraction' => $price_insights_result->getPriceInsights()->getPredictedConversionsChangeFraction(),
+					'effectiveness'                    => '', // $price_insights_result->getPriceInsights()->getEffectiveness() is not returning error, returning null.
```

Comment:
@joemcgill  Added note here `$price_insights_result->getPriceInsights()->getEffectiveness() is not returning error, returning null.`

Reply:
We'll likely need to modify the query to add an additional `price_insights.effectiveness` column, looking at [these docs](https://developers.google.com/shopping-content/reference/rest/v2.1/reports/search#priceinsights). We can handle that in #2911 .